### PR TITLE
meta-iotqa: add basic I2C bus testing

### PR DESCRIPTION
**Pull request by nvelux**
Signed-off-by: nvelux <nagarajagopalx.velu@intel.com>
test case for data sending and receiving to rs232 port

### CODE REVIEW

**Comment by nvelux** [comment]
Please help to test this build.

**Comment by okartau** [comment]
ok to test

**Comment by nvelux** [comment]
rs232 test case for manifest file, please test.

**Comment by okartau** [comment]
test this please,
(tester went offline during testing)

**Comment by okartau** [comment]
Error not shown in main console log, is hidden in detailed logfile
test_aft.minnowboardturbot.refkit-image-gateway.log:

File "iottest/oeqa/runtime/peripherals/mraa/mraa_rs232_cmd.py", line 10
    ...
      ^
IndentationError: expected an indented block

**Comment by nvelux** [comment]
I have uploaded I2C validation testing instead of rs232, 
RS232 need hardware loopback work for TX and RX, please ignore rs232 test now.
please test I2C testing.

**Comment by okartau** [comment]
test this please

**Comment by nvelux** [comment]
I have added i2c file in manifest , now you can do i2c testing.
please test.

**Comment by okartau** [comment]
test this please
CI master was restarted, seems job starting fail close to this time

**Comment by nvelux** [comment]
error corrected , please test.

**Comment by nvelux** [comment]
@okartau  it's giving error in different path, not in my script.
please test it.

File: meta-iotqa/lib/oeqa/runtime/peripherals/mraa/mraa_i2c.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    def test_mraa_i2c_list(self):
+        (status, output) = self.target.run("mraa-i2c list")
+        self.assertEqual(status, 0, msg="Error messages: %s" % output)
+
+    def test_mraa_i2c_detectbus(self):
+        (status, output) = self.target.run("mraa-i2c detect bus")
+        self.assertEqual(status, 0, msg="Error messages: %s" % output)
+
+    def test_i2c_value(self):
+        (status, output) = self.target.run("mraa-i2c set 1 0x02 0x03 0x04 > /tmp/log.txt")
```
Comment (mythi) [inline meta-iotqa/lib/oeqa/runtime/peripherals/mraa/mraa_i2c.py]: please add a comment to explain the purpose of this test and how the bus address, register, and value got chosen

File: meta-iotqa/lib/oeqa/runtime/peripherals/mraa/mraa_i2c.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        (status, output) = self.target.run("cat /tmp/log.txt|grep 'Value = 04'")
```
Comment (mythi) [inline meta-iotqa/lib/oeqa/runtime/peripherals/mraa/mraa_i2c.py]: Useless use of cat. You could ```grep 'Value = 04' /tmp/log.txt``` directly.

**Comment by nvelux** [comment]
i have done changes as per your suggestion for grep to find slave value and added comments.
please test it.

**Comment by mythi** [comment]
How about the question about the set parameters?

**Comment by nvelux** [comment]
mraa-i2c set command need 4 parameter to set i2c bus for slave, Device, Register and Value to avoid command running fail.
I know that there is no slave connected with i2c master, but want to pass data value and test passing value to slave is correct or not.

**Comment by mythi** [comment]
I don't have mraa-i2c installed right now so can you copy here the output of ```mraa-i2c set 1 0x02 0x03 0x04```

**Comment by nvelux** [comment]
Device 02, Register = 03, Value = 04

**Comment by mythi** [comment]
It seems the ```test_i2c_value``` is a rather useless test. Currently it returns success if mraa-i2c prints the parameters you have passed to it.

Until we have a way to check the value setting really succeeds, I'd suggest we drop  ```test_i2c_value```

**Comment by nvelux** [comment]
test_i2c_value removed and please test it.

**Comment by nvelux** [comment]
please merge the code to master branch.

**Comment by mythi** [comment]
@nvelux if this this ready from your perspective, please change the title to indicate that (i.e., not a test-build anymore

**Comment by nvelux** [comment]
Please remove the test build label and merge to master branch.

**Comment by mythi** [comment]
please rebase

**Comment by nvelux** [comment]
I have done rebase.

**Comment by mythi** [comment]
it's not done right it seems. "This branch has conflicts that must be resolved"

**Comment by nvelux** [comment]
it was completed without conflicts, please check once and merge with master branch.

**Comment by mythi** [comment]
it's not possible to merge it because of the errors. see what github says (and look at the commit too)

**Comment by okartau** [comment]
this PR seems to contain multiple changes for unrelated files, like Jenkinsfile and docker/*build* scripts,
showing changes that have been merged already in other PRs,
It seems this PR is not correctly rebased to latest head, otherwise it would not show such changes.

**Comment by nvelux** [comment]
please merge with master branch.

**Comment by mythi** [comment]
one more feedback: the commit message should be improved (+it has a typo varsion).

```meta-iotqa: add basic I2C bus testing``` and then in the message body you explain the rationale

**Comment by nvelux** [comment]
corrected, please merge with master branch.

**Pull request merged**